azure/application-gateway/advanced/outputs.tf

```diff
@@ -31,4 +31,14 @@ output "frontend_ip_configuration" {
 output "identity_id" {
   description = "Identity of the AppGW if KV is used."
   value       = azurerm_user_assigned_identity.main_gateway[0].id
+}
+
+output "private_ip" {
+  description = "Private IP Address"
+  value = var.private_ip
```

Comment:
Private IP should be present in ipconfig under `azurerm_application_gateway.main.frontend_ip_configuration[*].private_ip_address`

May be able to collect this with `one(azurerm_application_gateway.main.frontend_ip_configuration[*].private_ip_address)` but just in case I've put a suggestion for something that's worked for me before (probably more complex than needs to be)

```
  value = one([
    for ipconfig in azurerm_application_gateway.main.frontend_ip_configuration[*]
    : ipconfig.private_ip_address
    if ipconfig.private_ip_address != ""
  ])
```